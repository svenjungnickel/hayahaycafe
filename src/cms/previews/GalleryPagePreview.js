import React from 'react';
import PropTypes from 'prop-types';
import { GalleryPageTemplate } from '../../templates/GalleryPage';

const GalleryPagePreview = ({ entry, widgetFor }) => (
    <GalleryPageTemplate
        title={entry.toJS().data.title}
        subtitle={entry.toJS().data.subtitle}
        headerImage={entry.toJS().data.headerImage}
        content={widgetFor('body')}
        images={entry.toJS().data.images}
    />
);

GalleryPagePreview.propTypes = {
    entry: PropTypes.shape({
        toJS: PropTypes.func,
    }).isRequired,
    widgetFor: PropTypes.func.isRequired,
};

export default GalleryPagePreview;
