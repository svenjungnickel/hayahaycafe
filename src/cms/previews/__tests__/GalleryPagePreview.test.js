import React from 'react';
import renderer from 'react-test-renderer';
import GalleryPagePreview from '../GalleryPagePreview';
import GalleryPageData from '../../../__fixtures__/GalleryPageData';

jest.mock('../../../components/Header');
jest.mock('../../../components/Separator');
jest.mock('../../../components/Image');

describe('GalleryPagePreview', () => {
    it('renders without entry throws prop type error', () => {
        const renderStartPagePreview = () => {
            const widgetFor = () => {};

            const component = <GalleryPagePreview widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderStartPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders without widgetFor throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: () => {},
            };

            const component = <GalleryPagePreview entry={entry} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid entry throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: 123,
            };
            const widgetFor = () => {};

            const component = <GalleryPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid widgetFor throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: () => {},
            };
            const widgetFor = {};

            const component = <GalleryPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const entry = {
            toJS: () => {
                return {
                    data: GalleryPageData,
                };
            },
        };
        const widgetFor = () => 'content';

        const component = <GalleryPagePreview entry={entry} widgetFor={widgetFor} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
