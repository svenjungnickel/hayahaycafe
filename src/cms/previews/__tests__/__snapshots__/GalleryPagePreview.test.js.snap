// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPagePreview renders correctly 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="gallery container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="gallery__image col-lg-4 col-md-6 col-sm-12 col-12"
          data-cy="galleryImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
        <div
          className="gallery__image col-lg-4 col-md-6 col-sm-12 col-12"
          data-cy="galleryImage"
        >
          <div
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="src"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
